test/parallel_api/algorithm/alg.modifying.operations/replace_copy.pass.cpp

```diff
@@ -109,6 +109,135 @@ struct test_non_const
     }
 };
 
+struct not_implicitly_convertible
+{
+    explicit not_implicitly_convertible(int v) {}
+};
+
+template <typename It, typename DestIt, typename Void = void>
+struct is_replace_copy_well_formed : std::false_type {};
+
+template <typename It, typename DestIt>
+struct is_replace_copy_well_formed<It, DestIt,
+                                   std::void_t<decltype(oneapi::dpl::replace_copy(oneapi::dpl::execution::seq,
+                                                                                  std::declval<It>(),
+                                                                                  std::declval<It>(),
+                                                                                  std::declval<DestIt>(),
+                                                                                  {2},
+                                                                                  {3}))>> : std::true_type {};
+
+template <typename It, typename DestIt, typename Void = void>
+struct is_replace_copy_if_well_formed : std::false_type {};
+
+template <typename It, typename DestIt>
+struct is_replace_copy_if_well_formed<It, DestIt,
+                                      std::void_t<decltype(oneapi::dpl::replace_copy_if(oneapi::dpl::execution::seq,
+                                                                                        std::declval<It>(),
+                                                                                        std::declval<It>(),
+                                                                                        std::declval<DestIt>(),
+                                                                                        int{}, // actually, some callable should be here but since the function does not have any constraints any type will work
+                                                                                        {3}))>> : std::true_type {};
+
+constexpr void test_default_template_argument_from_output_iterator()
+{
+    static_assert(is_replace_copy_well_formed<std::vector<not_implicitly_convertible>::iterator, std::vector<int>::iterator>::value,
+                  "Positive: The default argument of replace_copy shall be taken from output iterator");
+    static_assert(!is_replace_copy_well_formed<std::vector<int>::iterator, std::vector<not_implicitly_convertible>::iterator>::value,
+                  "Negative: The default argument of replace_copy shall be taken from output iterator");
+    static_assert(is_replace_copy_if_well_formed<std::vector<not_implicitly_convertible>::iterator, std::vector<int>::iterator>::value,
+                  "Positive: The default argument of replace_copy_if shall be taken from output iterator");
+    static_assert(!is_replace_copy_if_well_formed<std::vector<int>::iterator, std::vector<not_implicitly_convertible>::iterator>::value,
+                  "Positive: The default argument of replace_copy_if shall be taken from output iterator");
```

Comment:
The "Positive" and "Negative" here are confusing for me (and the one on like 150 should be "Negative" I think).

I'd prefer something more like:

"replace_copy_if call is *NOT* well-formed for valid list_initialization argument"
and
"replace_copy_if call is well-formed for *INVALID* list_initialization argument" 

Or something similar so that it is clear to the user reading the error message what has gone wrong without looking at the tests.